src/libcore/cell.rs

```diff
@@ -521,6 +567,20 @@ impl<T: Default> Cell<T> {
 #[unstable(feature = "coerce_unsized", issue = "27732")]
 impl<T: CoerceUnsized<U>, U> CoerceUnsized<Cell<U>> for Cell<T> {}
 
+#[unstable(feature = "as_cell", issue="43038")]
+impl<T, I> Index<I> for Cell<[T]>
```

Comment:
Trait implementations are insta-stable, so this needs a review from @rust-lang/libs

Reply:
Sure it is, make a `&Cell<[T; N]>` and unsize it.